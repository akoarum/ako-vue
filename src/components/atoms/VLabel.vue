<script lang="ts">
import Vue, { CreateElement, VNode } from 'vue'

export default Vue.extend({
  props: {
    span: { type: Boolean }
  },
  render(createElement: CreateElement): VNode {
    return createElement(
      this.span ? 'span' : 'label',
      { class: 'vLabel' },
      this.$slots.default
    )
  }
})
</script>

<style lang="scss" scoped>
.vLabel {
  display: inline-block;
  font-weight: bold;

  @include media-min {
    font-size: $fontSize-pc;
  }

  @include media-max {
    font-size: $fontSize-sp;
  }
}
</style>
