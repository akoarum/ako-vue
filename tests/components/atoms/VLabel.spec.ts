import { shallowMount } from '@vue/test-utils'
import VLabel from '~/components/atoms/VLabel.vue'

describe('VLabel', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = shallowMount(VLabel, {
      propsData: {},
      slots: {
        default: 'ラベル'
      }
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('rendering', () => {
    it('通常はlabel要素でレンダリング', () => {
      expect(wrapper.is('label')).toBe(true)
    })

    it('props:trueの場合はspan要素でレンダリング', () => {
      wrapper.setProps({ span: true })
      expect(wrapper.is('span')).toBe(true)
    })
  })
})
