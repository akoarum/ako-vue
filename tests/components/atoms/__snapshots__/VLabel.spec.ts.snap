// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VLabel initialized snapshot 1`] = `
<label
  class="vLabel"
>
  ラベル
</label>
`;
